smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/utils/AuthUtils.kt

```diff
@@ -0,0 +1,39 @@
+package software.amazon.smithy.swift.codegen.utils
+
+import software.amazon.smithy.model.knowledge.ServiceIndex
+import software.amazon.smithy.model.traits.HttpBearerAuthTrait
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.config.DefaultProvider
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.swiftmodules.SmithyHTTPAuthTypes
+
+open class AuthUtils(
+    private val ctx: ProtocolGenerator.GenerationContext
+) {
+    val authSchemesDefaultProvider = DefaultProvider(
+        { getModeledAuthSchemesSupportedBySDK(ctx, it) },
+        isThrowable = false,
+        isAsync = false
+    )
+
+    fun getModeledAuthSchemesSupportedBySDK(
+        ctx: ProtocolGenerator.GenerationContext,
+        writer: SwiftWriter,
+    ): String {
+        val effectiveAuthSchemes = ServiceIndex(ctx.model).getEffectiveAuthSchemes(ctx.service)
+        var authSchemeList = arrayOf<String>()
+
+        if (effectiveAuthSchemes.contains(HttpBearerAuthTrait.ID)) {
+            authSchemeList += writer.format("\$N()", SmithyHTTPAuthTypes.BearerTokenAuthScheme)
+        }
+
+        authSchemeList = addAdditionalSchemes(writer, authSchemeList)
+
+        return "[${authSchemeList.joinToString(", ")}]"
+    }
+
+    open fun addAdditionalSchemes(writer: SwiftWriter, authSchemeList: Array<String>): Array<String> {
```

Comment:
Change function signature to use Lists instead of Arrays

`open fun addAdditionalSchemes(writer: SwiftWriter, authSchemeList: MutableList<String>): List<String>`

Reply:
Changed as suggested!